src/main/scala/outwatch/dom/DomTypes.scala

```diff
@@ -26,14 +26,21 @@ private[outwatch] object DomTypesBuilder {
     type Attribute[T, _] = AttributeBuilder[T]
     type Property[T, _] = PropertyBuilder[T]
 
-    def encode[V](codec: Codec[V, _]): V => Attribute.Value = codec match {
+    def encodeProperty[V](codec: Codec[V, _]): V => Attribute.Value = value => {
+      // codec.encode may encode to null values
+      codec.encode(value) match {
+        case b: Boolean => b
+        case null => null
+        case v => v.toString
+      }
+    }
+
+    def encodeAttribute[V](codec: Codec[V, String]): V => Attribute.Value = codec match {
       //The BooleanAsAttrPresenceCodec does not play well with snabbdom. it
       //encodes true as "" and false as null, whereas snabbdom needs true/false
       //of type boolean (not string) for toggling the presence of the attribute.
       case _: BooleanAsAttrPresenceCodec.type => identity
-      case _: BooleanAsIsCodec.type => identity
-      // codec.encode may encode to null values
-      case _ => (codec.encode _) andThen (v => if (v != null) v.toString else null)
+      case _ => codec.encode _
```

Comment:
This should work too, but I still don't understand why having a single `encode()` as before wouldn't work. As far as I can tell `BooleanAsIsCodec` is only used for props. Do you have an example where it doesn't work as expected?

Reply:
Fair point.

Reply:
still we need to do it differently for attributes and properties.